src/graphic/Fast3D/gfx_opengl.cpp

```diff
@@ -554,9 +581,13 @@ static struct ShaderProgram* gfx_opengl_create_and_load_new_shader(uint64_t shad
     }
 
     if (cc_features.opt_alpha && cc_features.opt_noise) {
+        append_line(
+            fs_buf, &fs_len,
+            "texel.a = alphaDither(random(vec3(floor(gl_FragCoord.xy * noise_scale), float(frame_count))), texel.a);");
+    } else if (cc_features.opt_noise) {
         append_line(fs_buf, &fs_len,
-                    "texel.a *= floor(clamp(random(vec3(floor(gl_FragCoord.xy * noise_scale), float(frame_count))) + "
-                    "texel.a, 0.0, 1.0));");
+                    "texel.rgb = colorDither(random(vec3(floor(gl_FragCoord.xy * noise_scale), float(frame_count))), "
+                    "texel.rgb);");
     }
```

Comment:
At least from what I can see in MM, it is possible for aplha dithering and color dithering to be applied at the same time. This `if / if else` does not allow that currently.

We may even need to split `opt_noise` into two values, or something so that we can control which is used.